collated/main/Jeremy.md

```diff
@@ -0,0 +1,1529 @@
+                multipleAndOrCluster = true;
+                break;
+            }
+            previousString = str;
+        }
+        return multipleAndOrCluster;
+    }
+}
+```
+###### \java\seedu\address\logic\parser\ParserUtil.java
+``` java
+    /**
+     * Parses a {@code Optional<String> remark} into an {@code Optional<Remark>} if {@code remark} is present.
+     * See header comment of this class regarding the use of {@code Optional} parameters.
+     */
+    public static Optional<Remark> parseRemark(Optional<String> remark) throws IllegalValueException {
+        requireNonNull(remark);
+        return remark.isPresent() ? Optional.of(new Remark(remark.get())) : Optional.empty();
+    }
+```
+###### \java\seedu\address\logic\parser\RemarkCommandParser.java
+``` java
+/**
+ * Parses input arguments and creates a new RemarkCommand object
+ */
+public class RemarkCommandParser implements Parser<RemarkCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the RemarkCommand
+     * and returns an RemarkCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RemarkCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_REMARK);
+
+        Index indexInput;
+
+        if (!isPrefixPresent(argMultimap, PREFIX_REMARK)) {
+            throw new ParseException(String.format(
+                    MESSAGE_INVALID_COMMAND_FORMAT, RemarkCommand.MESSAGE_USAGE));
+        }
+        try {
+            indexInput = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException ive) {
+            throw new ParseException(ive.getMessage(), ive);
+        }
+
+        String remark = argMultimap.getValue(PREFIX_REMARK).orElse("");
+
+        return new RemarkCommand(indexInput, new Remark(remark));
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean isPrefixPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+
+}
+```
+###### \java\seedu\address\model\AddressBook.java
+``` java
+    public ObservableList<ReadOnlyPerson> getPersonListSortByNameAscending() {
+        return persons.asObservableListSortedByNameAsc();
+    }
+
+    public ObservableList<ReadOnlyPerson> getPersonListSortByNameDescending() {
+        return persons.asObservableListSortedByNameDsc();
+    }
+
+    public ObservableList<ReadOnlyPerson> getPersonListReversed() {
+        return persons.asObservableListReversed();
+    }
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /**
+     * @return an unmodifiable view of the list of ReadOnlyPerson that has nonNull name,
```

Comment:
The start of the header comment does not follow the convention (i.e. first sentence should be something like `Selects next line …` instead of `Select next line` or `This method is for selecting …`)